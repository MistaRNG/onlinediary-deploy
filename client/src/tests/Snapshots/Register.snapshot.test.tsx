import React from 'react';
import { render } from '@testing-library/react';
import Register from '../../features/auth/Register';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import authReducer from '../../features/auth/authSlice';
import { BrowserRouter as Router } from 'react-router-dom';
import { describe, it, expect } from '@jest/globals';

const mockStore = configureStore({
  reducer: {
    auth: authReducer,
  },
});

describe('Register Component Snapshot Tests', () => {
  it('should render the Register component correctly', () => {
    // ARRANGE & ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
