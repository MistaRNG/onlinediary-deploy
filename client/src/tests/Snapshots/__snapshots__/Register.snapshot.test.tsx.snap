// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Snapshot Tests should render the Register component correctly 1`] = `
<DocumentFragment>
  <h1>
    Register
  </h1>
  <form>
    <div
      class="bp4-form-group"
    >
      <label
        class="bp4-label"
      >
        Username 
        <span
          class="bp4-text-muted"
        />
      </label>
      <div
        class="bp4-form-content"
      >
        <div
          class="bp4-input-group bp4-large"
        >
          <input
            class="bp4-input"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      class="bp4-form-group"
    >
      <label
        class="bp4-label"
      >
        Password 
        <span
          class="bp4-text-muted"
        />
      </label>
      <div
        class="bp4-form-content"
      >
        <div
          class="bp4-input-group bp4-large"
        >
          <input
            autocomplete="on"
            class="bp4-input"
            style="padding-right: 0px;"
            type="password"
          />
          <span
            class="bp4-input-action"
          >
            <button
              class="bp4-button bp4-minimal bp4-intent-warning"
              type="button"
            >
              <span
                aria-hidden="true"
                class="bp4-icon bp4-icon-lock"
                icon="lock"
              >
                <svg
                  data-icon="lock"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M13.96 7H12V3.95C12 1.77 10.21 0 8 0S4 1.77 4 3.95V7H1.96c-.55 0-.96.35-.96.9v6.91c0 .54.41 1.19.96 1.19h12c.55 0 1.04-.65 1.04-1.19V7.9c0-.55-.49-.9-1.04-.9zM6 7V3.95c0-1.09.9-1.97 2-1.97s2 .88 2 1.97V7H6z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="bp4-form-group"
    >
      <label
        class="bp4-label"
      >
        Confirm Password 
        <span
          class="bp4-text-muted"
        />
      </label>
      <div
        class="bp4-form-content"
      >
        <div
          class="bp4-input-group bp4-large"
        >
          <input
            autocomplete="on"
            class="bp4-input"
            style="padding-right: 0px;"
            type="password"
          />
          <span
            class="bp4-input-action"
          >
            <button
              class="bp4-button bp4-minimal bp4-intent-warning"
              type="button"
            >
              <span
                aria-hidden="true"
                class="bp4-icon bp4-icon-lock"
                icon="lock"
              >
                <svg
                  data-icon="lock"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M13.96 7H12V3.95C12 1.77 10.21 0 8 0S4 1.77 4 3.95V7H1.96c-.55 0-.96.35-.96.9v6.91c0 .54.41 1.19.96 1.19h12c.55 0 1.04-.65 1.04-1.19V7.9c0-.55-.49-.9-1.04-.9zM6 7V3.95c0-1.09.9-1.97 2-1.97s2 .88 2 1.97V7H6z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
    <button
      class="bp4-button bp4-large"
      type="submit"
    >
      <span
        class="bp4-button-text"
      >
        Register
      </span>
    </button>
  </form>
  <div
    class="user-link"
  >
    Already a member? 
    <a
      href="/login"
    >
      Login
    </a>
  </div>
</DocumentFragment>
`;
